Sources/MongoSwift/MongoConnectionString.swift

```diff
@@ -135,18 +134,29 @@ public struct MongoConnectionString: Codable, LosslessStringConvertible {
                 self.type = .ipLiteral
             } else {
                 let parts = hostAndPort.components(separatedBy: ":")
-                self.host = String(parts[0])
                 guard parts.count <= 2 else {
                     throw MongoError.InvalidArgumentError(
                         message: "expected only a single port delimiter ':' in \(hostAndPort)"
                     )
                 }
+
+                let host = parts[0]
+                if host.hasSuffix(".sock") {
+                    self.host = try host.getPercentDecoded(forKey: "UNIX domain socket")
+                    self.type = .unixDomainSocket
+                } else if host.isIPV4() {
+                    self.host = host
+                    self.type = .ipv4
+                } else {
+                    self.host = try host.getPercentDecoded(forKey: "hostname")
```

Comment:
The rule for reg-name (i.e. a non-IPLiteral/IPv4 host) is:
```
reg-name    = *( unreserved / pct-encoded / sub-delims )
```
so we want to support percent decoding here.